parsers/AX.py

```diff
@@ -75,12 +75,15 @@ def formated_data(
     """Format data to Electricity Map standards."""
     data_list = fetch_data(session, logger)
     data_list.reverse()
-    date = datetime.now(timezone(TIME_ZONE)).replace(
+    date_time = datetime.now(timezone(TIME_ZONE))
+    date = date_time.replace(
         hour=int(data_list[0]["time"].split(":")[0]),
         minute=int(data_list[0]["time"].split(":")[1]),
         second=0,
         microsecond=0,
     )
+    if date > date_time:
```

Comment:
I'm wondering about the effect of this. Will this mean that at midnight on a day, the datapoint will be attributed to yesterday ?

Reply:
Since we are just getting the hours and minutes back from the iFrame we run the risk of assigning the past values to a future date at midnight (local time) if we fetch data after midnight but before the iFrame has updated.

So for example if we fetch data December 5 00:01 and the iFrame updates at 00:02 we would assign the data for the past hour starting December 5 23:45 and then backwards 24 hours. This is obviously not what we want.

So if the date we get from the parser is grater than the date when we run the parser we shift the  date back one day (which should be the correct date).

Reply:
Ok makes sense then to do this!